relay-general/src/store/normalize.rs

```diff
@@ -90,20 +90,25 @@ pub fn is_valid_platform(platform: &str) -> bool {
     VALID_PLATFORMS.contains(&platform)
 }
 
-pub fn normalize_dist(dist: &mut Option<String>) {
-    let mut erase = false;
+pub fn normalize_dist(distribution: &mut Annotated<String>) {
+    let dist = distribution.value_mut();
     if let Some(val) = dist {
         if val.is_empty() {
-            erase = true;
-        }
-        let trimmed = val.trim();
-        if trimmed != val {
-            *val = trimmed.to_string()
+            *dist = None;
+        } else {
+            let trimmed = val.trim();
+            // Validate the length of the distribution name and make sure to attach the error in
+            // case of the issues.
+            if bytecount::num_chars(trimmed.as_bytes()) > MaxChars::Distribution.limit() {
+                *distribution = Annotated::from_error(
+                    Error::new(ErrorKind::ValueTooLong),
+                    Some(Value::String(val.to_owned())),
+                )
```

Comment:
Instead of creating a new `Annotated` object, does it make sense to just add the error to the meta? Something like the following:
```suggestion
                distribution.meta_mut().add_error(...);
```
In this case, you'd still have the initial string there which is too long, so I'm not sure if this suggestion makes sense.

Reply:
I've rewritten this using `apply` method